modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -831,6 +831,8 @@ class CV_EXPORTS_W QRCodeDetector
 }
 
 #include "opencv2/objdetect/detection_based_tracker.hpp"
+#ifndef WINRT
 #include "opencv2/objdetect/face.hpp"
+#endif
```

Comment:
Conditional compilation in public headers are not allowed.

We should have `#ifdef HAVE_OPENCV_DNN` in .cpp files only and provide stubs in `#else` section.

Reply:
> provide stubs in #else section.

What is stubs? How to pribide it?

Reply:
Class/methods/functions stubs with error messages like this: https://github.com/opencv/opencv/blob/4.5.5/modules/calib3d/src/circlesgrid.cpp#L425

Reply:
Thanks, I added it, however I have no confidence. If there are any problems, please point them out.